### ISSUE TITLE
Add Vaultwarden for password management

### ISSUE THREAD

**Issue by chutch3** [chutch3/selfhosted.sh#64]
## Summary

Add [Vaultwarden](https://github.com/dani-garcia/vaultwarden) as a self-hosted password manager compatible with Bitwarden clients. Essential security infrastructure for managing credentials across the homelab and personal devices.

### Security & Convenience
- **Password Management**: Secure vault for all passwords and credentials
- **Cross-Platform**: Desktop, mobile, and browser extensions
- **Bitwarden Compatible**: Use official Bitwarden clients
- **2FA Support**: TOTP, U2F, Duo, YubiKey
- **Secure Sharing**: Share passwords with family/team members
- **Self-Hosted**: Full control over sensitive data
- **Autofill**: Browser and mobile autofill capabilities

### Homelab Integration
- Store credentials for all homelab services
- Manage SSH keys and API tokens
- Share service credentials with family members
- Secure backup of critical passwords

### Storage
- **Database**: SQLite (default) or PostgreSQL/MySQL
- **Application Data**: <1GB typically
- **Attachments**: Variable (if used)
- **Storage Type**: Local storage with regular backups

### Memory
- **Minimum**: 256MB RAM
- **Recommended**: 512MB-1GB RAM

### CPU
- Very lightweight, minimal CPU usage

### Security Requirements
- **HTTPS Required**: Must use SSL (Bitwarden requirement)
- **Strong Master Password**: User responsibility
- **Regular Backups**: Critical for disaster recovery

## Implementation Plan

1. **Create Vaultwarden Stack**
   - [ ] Create `stacks/apps/vaultwarden/` directory
   - [ ] Create `docker-compose.yml` with configuration:
     - Traefik labels with SSL (`vault.${BASE_DOMAIN}`)
     - Persistent volumes for data and attachments
     - Admin token for admin panel access
     - WebSocket support for live sync
     - Deployment on storage node

2. **Security Configuration**
   - [ ] Generate admin token: `openssl rand -base64 48`
   - [ ] Configure HTTPS enforcement
   - [ ] Set up email notifications (optional)
   - [ ] Configure 2FA options
   - [ ] Set signup restrictions (disable public signup)

3. **Backup Strategy**
   - [ ] Configure automated backups of data directory
   - [ ] Test backup restoration process
   - [ ] Document backup location and frequency
   - [ ] Consider encryption for backup files

4. **Client Setup**
   - [ ] Install Bitwarden browser extension
   - [ ] Configure mobile apps to use self-hosted instance
   - [ ] Test autofill and sync functionality
   - [ ] Import existing passwords (if applicable)

5. **Documentation**
   - [ ] Add `.env.example` variables
   - [ ] Document admin panel access
   - [ ] Create backup/restore procedures
   - [ ] Add to Homepage dashboard

## Configuration

**Environment Variables:**
```bash
# Vaultwarden Configuration
VAULTWARDEN_ADMIN_TOKEN=<generated_token>
VAULTWARDEN_DOMAIN=https://vault.${BASE_DOMAIN}
VAULTWARDEN_SMTP_HOST=smtp.example.com  # Optional
VAULTWARDEN_SMTP_FROM=vaultwarden@${BASE_DOMAIN}
VAULTWARDEN_SIGNUPS_ALLOWED=false  # Disable after initial setup
```

**Critical Paths:**
- Data: `/data` (contains database and configuration)
- Attachments: `/data/attachments`
- Admin Panel: `https://vault.${BASE_DOMAIN}/admin`

## Testing Plan

1. **Verify Deployment:**
   ```bash
   docker service ls | grep vaultwarden
   docker service ps vaultwarden
   ```

2. **Test Web Vault:**
   - Access web vault at `https://vault.${BASE_DOMAIN}`
   - Create test account
   - Add test password entry
   - Verify WebSocket connection (live sync)

3. **Test Browser Extension:**
   - Install Bitwarden extension
   - Configure custom server URL
   - Login and verify sync
   - Test autofill on a website

4. **Test Admin Panel:**
   - Access `/admin` with admin token
   - Review diagnostics
   - Check user list
   - Verify configuration

5. **Test Backup/Restore:**
   - Create backup of `/data` directory
   - Verify backup integrity
   - Test restoration process (in staging)

## Security Considerations

**Critical:**
- ⚠️ **HTTPS is mandatory** - Bitwarden clients require SSL
- ⚠️ **Strong admin token** - Protects admin panel
- ⚠️ **Regular backups** - Data loss = lost passwords
- ⚠️ **Secure master password** - User's responsibility
- ⚠️ **Disable signups** - After creating accounts

**Recommendations:**
- Enable 2FA for all accounts
- Regular security audits of stored credentials
- Monitor failed login attempts
- Keep Vaultwarden updated (use Watchtower)
- Store backup encryption keys separately

## Backup Strategy

```bash
# Backup script example
BACKUP_DIR=/backups/vaultwarden
DATE=$(date +%Y%m%d_%H%M%S)

# Backup data directory
docker cp vaultwarden:/data $BACKUP_DIR/vaultwarden-$DATE

# Encrypt backup (optional but recommended)
tar czf - $BACKUP_DIR/vaultwarden-$DATE | \
  gpg --encrypt --recipient your@email.com > \
  $BACKUP_DIR/vaultwarden-$DATE.tar.gz.gpg

# Upload to Backblaze (integrate with existing backup)
rclone copy $BACKUP_DIR/vaultwarden-$DATE.tar.gz.gpg \
  backblaze:homelab-backups/vaultwarden/
```

## Resources

- GitHub: https://github.com/dani-garcia/vaultwarden
- Documentation: https://github.com/dani-garcia/vaultwarden/wiki
- Bitwarden Clients: https://bitwarden.com/download/
- Technology: Rust
- License: AGPL-3.0

## Acceptance Criteria

- [ ] Vaultwarden deployed with HTTPS enabled
- [ ] Admin panel accessible and secured
- [ ] Test account created and working
- [ ] Browser extension connected and syncing
- [ ] Mobile app connected and syncing
- [ ] Backup strategy implemented and tested
- [ ] Public signups disabled
- [ ] Service included in Homepage dashboard
- [ ] Documentation updated with setup instructions

## Additional Notes

- Vaultwarden is a lightweight, unofficial Bitwarden-compatible server
- Uses significantly less resources than official Bitwarden server
- Fully compatible with all official Bitwarden clients
- Some enterprise features not available (SSO, directory sync)
- Consider exposing externally for mobile access (with strong security)
- Can migrate to/from official Bitwarden server if needed
